acorn/src/statement.js

```diff
@@ -72,6 +72,39 @@ pp.isAsyncFunction = function() {
      !(isIdentifierChar(after = this.input.charCodeAt(next + 8)) || after > 0xd7ff && after < 0xdc00))
 }
 
+pp.isUsingKeyword = function(isAwaitUsing) {
+  if (this.options.ecmaVersion < 17 || !this.isContextual(isAwaitUsing ? "await" : "using"))
+    return false
+
+  skipWhiteSpace.lastIndex = this.pos
+  let skip = skipWhiteSpace.exec(this.input)
+  let next = this.pos + skip[0].length
+
+  if (lineBreak.test(this.input.slice(this.pos, next))) return false
+
+  if (isAwaitUsing) {
+    let usingEndPos = next + 5 /** using */
+    if (this.input.slice(next, usingEndPos) !== "using") return false
+    skipWhiteSpace.lastIndex = 0
+    let skipAfterUsing = skipWhiteSpace.exec(this.input.slice(usingEndPos))
+    if (skipAfterUsing) {
+      let nextAfterUsing = usingEndPos + skipAfterUsing[0].length
+      if (lineBreak.test(this.input.slice(usingEndPos, nextAfterUsing))) return false
+    }
+  }
+
+  let ch = this.input.charCodeAt(next)
+  return isIdentifierStart(ch, true) || ch === 92 // '\'
+}
+
+pp.isAwaitUsingDeclaration = function() {
```

Comment:
The naming seemed inconsistent to me.
The next method name is `isUsing`, so I think `isAwaitUsing` would be better here.

Reply:
updated: https://github.com/acornjs/acorn/pull/1369/commits/17b3f8513a960c86f2eeb8e4dc890729db9bfaac